Update add_missing_switch_cases code template

firephreek opened the pull request:
Updated code generation to 'throw UnimplementedError()' instead of returning 'null' and put the 'Handle this case' reminder message in the Error instead of an additional comment. 

This fix addresses issue [#53832](https://github.com/dart-lang/sdk/issues/53832). 

---

- [X] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/sdk/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.

Note that this repository uses Gerrit for code reviews. Your pull request will be automatically converted into a Gerrit CL and a link to the CL written into this PR. The review will happen on Gerrit but you can also push additional commits to this PR to update the code review.
</details>

firephreek committed:
Update code generation to 'throw UnimplementedError()' instead of returning 'null'.

'Handle this case' reminder message is now in the Error instead of a comment.

scheglov commented on pkg/analysis_server/lib/src/services/correction/dart/add_missing_switch_cases.dart:
```diff
@@ -70,11 +70,8 @@ class AddMissingSwitchCases extends ResolvedCorrectionProducer {
         builder.write(location.prefix);
         builder.write(lineIndent);
         builder.write(singleIndent);
-        builder.writeln('// TODO: Handle this case.');
```
Using `UnimplementedError` is seems a good idea to me.
But I will also keep the `// TODO ...` generating code.

scheglov commented on pkg/analysis_server/lib/src/services/correction/dart/add_missing_switch_cases.dart:
```diff
@@ -111,9 +108,9 @@ class AddMissingSwitchCases extends ResolvedCorrectionProducer {
-    DartEditBuilder builder,
-    List<MissingPatternPart> parts,
-  ) {
+      DartEditBuilder builder,
```
It looks that this code is not formatted.

scheglov commented:
Ah, there is also Gerrit CL.
I will repeat my comments there :-)

firephreek committed:
Re-added '// TODO' and removed inline message.
Kept use of single-quotes since inline message is gone.
Fixed way-ward formatting.

parlough commented:
Thanks for this contribution! Definitely makes sense :D
